hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -46,6 +52,8 @@
             parentToChildrenEventIds,
             eventMap);
 
+    Map<ByteBuffer, EventInfo> eventInfoMap = buildEventInfoMap(
```

Comment:
Why do this in view generation vs enrichment? I've always wondered why there's any logic in view generation - is there any benefit? The main problems being - it decentralizes (between view generation and enrichment) logic and makes this attribute only available at query time, while others are available at enrichment time.

Reply:
This is an interesting point, and I didn't think about it before making the change. 

>  I've always wondered why there's any logic in view generation - is there any benefit? 

To not have any logic in view-generation, all of the field in the views which are computed in view-gens should be moved upstream. This might bloat up somewhat the StructuredTrace packet. 

>  logic and makes this attribute only available at query time, while others are available at enrichment time
Maybe this is the reason, basically having a limited number of metrics at enrichment time and having distinction from query time attributes. 

I feel like continuing with the change, and move this particular field as enriched attribute if and when there is need.

Reply:
> To not have any logic in view-generation, all of the field in the views which are computed in view-gens should be moved upstream. This might bloat up somewhat the StructuredTrace packet.

In my opinion, any bloat of individual attributes (in this case, a single int) is worth the simplification of design - especially given that bloat already exists at rest once it lands in pinot. Doing things just in time means that accessing attributes needs to understand the implementation of the specific attribute and where/how it is populated. In this specific case too, it means that we need to do fairly significant processing in view generation, while my understanding is that similar processing has already happened in the enrichers that can be reused.

Reply:
> especially given that bloat already exists at rest once it lands in pinot. 

This is a separate thing, StructuredTrace packet can be have other consumers as well. 

However, I think what you are proposing makes sense, but then I have one question, if the processing logic moves out from view-generators to an enrichment job in trace-enricher (which is completely doable and optimises multiple things) why run view-generator services even?

Reply:
I think, at least in my mind, the ultimate goal is that view generation should be completely config driven. Whether projection ot materialized in enrichment, I think keeping all attribute-specific logic out of view generation would be a good thing. If the job could then be removed entirely (not familiar enough with pinot ingestion to speak to that), that would be another big improvement.

Reply:
Logged https://github.com/hypertrace/hypertrace-ingester/issues/139